Make `check_doc_links` fails with UnicodeDecodeError: 'utf-8' codec can't decode byte 0x96 in position 522: invalid start byte

**Describe the bug**

There are 2 problems when running `make check_doc_links` :

1. too many "third-party" Markdown files from project dependencies are scanned for hyperlinks which cause the process to take way too much time
2. if any of the Markdown files is not UTF-8 encoded, the check errors out


**To Reproduce**

Build the MLX UI (see [dashboard/origin-mlx/README.md](https://github.com/machine-learning-exchange/mlx/tree/main/dashboard/origin-mlx#setup)) which will create the `node_modules` folder. Then, at the project root level, run `make check_doc_links`.

```
[mlx] $ make check_doc_links

Checking for Markdown files here:

  **/*.md
  bootstrapper/catalog_upload.json

Traceback (most recent call last):
  File "/Users/ckadner/PycharmProjects/mlx_ckadner/tools/python/verify_doc_links.py", line 188, in <module>
    verify_doc_links()
  File "/Users/ckadner/PycharmProjects/mlx_ckadner/tools/python/verify_doc_links.py", line 151, in verify_doc_links
    file_line_text_url = [
  File "/Users/ckadner/PycharmProjects/mlx_ckadner/tools/python/verify_doc_links.py", line 154, in <listcomp>
    for (line, text, url) in get_links_from_md_file(file)
  File "/Users/ckadner/PycharmProjects/mlx_ckadner/tools/python/verify_doc_links.py", line 52, in get_links_from_md_file
    md_file_content = f.read()
  File "/usr/local/Cellar/python@3.9/3.9.7_1/Frameworks/Python.framework/Versions/3.9/lib/python3.9/codecs.py", line 322, in decode
    (result, consumed) = self._buffer_decode(data, self.errors, final)
UnicodeDecodeError: 'utf-8' codec can't decode byte 0x96 in position 522: invalid start byte
make: *** [check_doc_links] Error 1

```

--- a/tools/python/verify_doc_links.py
+++ b/tools/python/verify_doc_links.py
@@ -24,6 +24,11 @@ md_file_path_expressions = [
     "/bootstrapper/catalog_upload.json",
 ]
 
+excluded_paths = [
+    "node_modules",
+    "temp",
+]
+
 script_folder = abspath(dirname(__file__))
 project_root_dir = abspath(dirname(dirname(script_folder)))
 github_repo_master_path = "{}/blob/master".format(GITHUB_REPO.rstrip("/"))
@@ -40,16 +45,25 @@ def find_md_files() -> [str]:
         print("  " + path_expr.lstrip("/"))
     print("")
 
-    md_files_list_of_lists = [glob(project_root_dir + path_expr, recursive=True)
-                              for path_expr in md_file_path_expressions]
+    list_of_lists = [glob(project_root_dir + path_expr, recursive=True)
+                     for path_expr in md_file_path_expressions]
 
-    return sorted(list(itertools.chain(*md_files_list_of_lists)))
+    flattened_list = list(itertools.chain(*list_of_lists))
+
+    filtered_list = [path for path in flattened_list
+                     if not any(s in path for s in excluded_paths)]
+
+    return sorted(filtered_list)
 
 
-def get_links_from_md_file(md_file_path: str) -> [(int, str, str)]: # -> [(line, link_text, URL)]
+def get_links_from_md_file(md_file_path: str) -> [(int, str, str)]:  # -> [(line, link_text, URL)]
 
     with open(md_file_path, "r") as f:
-        md_file_content = f.read()
+        try:
+            md_file_content = f.read()
+        except ValueError as e:
+            print(f"Error trying to load file {md_file_path}")
+            raise e
 
     folder = relpath(dirname(md_file_path), project_root_dir)
 

